cpu/stm32_common/periph/uart.c

```diff
@@ -154,6 +154,41 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     return UART_OK;
 }
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+int uart_mode(uart_t uart, uart_data_bits_t data_bits, uart_parity_t parity,
+              uart_stop_bits_t stop_bits)
+{
+    assert(uart < UART_NUMOF);
+
+    if (parity) {
+        if (data_bits == UART_DATA_BITS_8) {
+            data_bits = USART_CR1_M;
+        }
+        else if (data_bits == UART_DATA_BITS_7) {
+            data_bits = UART_DATA_BITS_8;
+        }
+        else if (data_bits == UART_DATA_BITS_6) {
+            data_bits = UART_DATA_BITS_7;
+        }
+    }
+    if ((data_bits & UART_INVALID_MODE) || (parity & UART_INVALID_MODE)) {
+        return UART_NOMODE;
+    }
+
+#ifdef USART_CR1_M1
+    if (!(dev(uart)->ISR & USART_ISR_TC)) {
+        return UART_INTERR;
+    }
+    dev(uart)->CR1 &= ~(USART_CR1_UE | USART_CR1_TE);
+#endif
+
+    dev(uart)->CR2 |= stop_bits;
+    dev(uart)->CR1 |= (USART_CR1_UE | USART_CR1_TE | data_bits | parity);
```

Comment:
why tough the `UE` and `TE` bits here? Also as stated above, the data length and parity bits should be reset before setting them...

Reply:
My bad, I suggested that.  I also tested that, I am not sure if before can also me at the same time assuming the bits were cleared initially.

Reply:
It can be done at the same time.
The reason for this is the [stm32l0xxx](https://www.st.com/content/ccc/resource/technical/document/reference_manual/2f/b9/c6/34/28/29/42/d2/DM00095744.pdf/files/DM00095744.pdf/jcr:content/translations/en.DM00095744.pdf) page 802 states

> This bit can only be written when the USART is disabled (UE=0).

Reply:
Reworked `uart_mode` to first clear data and stop bits as also parity settings.